SS_objfunc.tpl

```diff
@@ -594,12 +621,42 @@ FUNCTION void evaluate_the_objective_function()
         f = abs(SzFreq_obs_hdr(iobs, 3));
         z1 = SzFreq_obs_hdr(iobs, 7);
         z2 = SzFreq_obs_hdr(iobs, 8);
-        SzFreq_like(SzFreq_LikeComponent(f, k)) -= SzFreq_sampleN(iobs) * SzFreq_obs(iobs)(z1, z2) * log(SzFreq_exp(iobs)(z1, z2));
+        int Sz_method = SzFreq_obs1(iobs, 1);  //  sizecomp method
+        int logL_method = Comp_Err_Sz(Sz_method);
+        temp_logL = 0.0;
+        switch (logL_method)
+        {
+          case 0:  //  multinomial
+          {
+            temp_logL += Comp_logL_multinomial( SzFreq_sampleN(iobs), SzFreq_obs(iobs)(z1, z2), SzFreq_exp(iobs)(z1, z2));
+            break;
+          }
+          case 1:  // dirichlet with theta*n
+          {
+             dirichlet_Parm = mfexp(selparm(Comp_Err_parmloc(Comp_Err_Sz2(Sz_method),1))) * SzFreq_sampleN(iobs);  //  theta * n
+             temp_logL -= gammln(dirichlet_Parm) - gammln( SzFreq_sampleN(iobs) + dirichlet_Parm );
+             temp_logL -= Comp_logL_Dirichlet( SzFreq_sampleN(iobs), dirichlet_Parm, SzFreq_obs(iobs)(z1, z2), SzFreq_exp(iobs)(z1, z2));
+             break;
+          }
+          case 2:  // dirichlet with beta
+          {
+             dirichlet_Parm = mfexp(selparm(Comp_Err_parmloc(Comp_Err_Sz2(Sz_method),1)));  //  beta
+             temp_logL -= gammln(dirichlet_Parm) - gammln( SzFreq_sampleN(iobs) + dirichlet_Parm );
+             temp_logL -= Comp_logL_Dirichlet( SzFreq_sampleN(iobs), dirichlet_Parm, SzFreq_obs(iobs)(z1, z2), SzFreq_exp(iobs)(z1, z2));
+             break;
+          }
+          case 3:  // MV  Tweedie
+          {
+            break;
```

Comment:
Would it make sense to stop with an error here, or does Tweedie for generalized size comp get flagged earlier? If you're planning on adding this option before merging, then no need.

Reply:
I was planning to add Tweedie after we update to ADMB 13.0.  For now, no one knows it is even here, so little risk.